### ISSUE TITLE
Ruby Support

### ISSUE THREAD

**Issue by sandstrom** [bytecodealliance/wit-bindgen#143]
Just wanted to open an issue here, about adding Ruby support. I really like Wasmtime and WebAssembly, and support for Ruby would be awesome! 💎

**Comment by jeffparsons** [comment]
Unfortunately the Ruby bindings for Wasmtime itself are third-party, and appear to be unmaintained. So I can't imagine effort being directed toward supporting it in wit-bindgen unless somebody revives that project first.

**Comment by jeffparsons** [comment]
I'd guess that whether or not the Bytecode Alliance will build and maintain their own Ruby bindings depends largely on whether any of the member organizations are heavily invested in Ruby already.

> Python, a similar language, is already officially supported.

That might actually make it _less_ likely that they'd be interested in Ruby bindings in the short term; they're already providing an example of integrating with that kind of language.

:man_shrugging: I'm just guessing here. I'd love to see Ruby bindings, too, because my employer uses a lot of Ruby and a lot of TypeScript. I reckon we could do some really cool stuff both on the server side itself and in sharing some code between server and client.

**Comment by sandstrom** [comment]
@jeffparsons If you'd be interested in working on something in the Ruby<>Wasmtime space in your spare time, we could probably sponsor some of the work via Github. But I'm guessing you already have a lot of things on your plate.

**Comment by jeffparsons** [comment]
> @jeffparsons If you'd be interested in working on something in the Ruby<>Wasmtime space in your spare time, we could probably sponsor some of the work via Github. But I'm guessing you already have a lot of things on your plate.

Good guess. :) It's definitely the kind of work I'd be interested in theoretically, but in practice `$LIFE` reasons (the positive kind) mean I have about zero hours free for that sort of thing.

It could be worth putting your feelers out into the Rust/Wasm/Ruby communities in general, on the off chance you might discover (a) people who might be suitable for that sort of work; and (b) other orgs who might be able to pitch in sponsorship.

**Comment by emilebosch** [comment]
Hey all, any status on this? Would love to implement this at some point.

**Comment by Mossaka** [comment]
I don't think it has any progress. With that said, feel free to go ahead and implement it, @emilebosch !

**Comment by sandstrom** [comment]
I've clarified my top comment, highlighting that this is about Ruby as a guest language (because that's what wit-bindgen is concerned about afaik).

If someone is looking to use Ruby as host language (to run WebAssembly code compiled down from other language, [see this excellent project](https://github.com/bytecodealliance/wasmtime-rb).
